### DISCUSSION TITLE
@Epic not visible in allure report when using parameterised tests

### DISCUSSION THREAD

**Discussion by MM-KR** [allure-framework/allure2#2068] [Questions & Support] [answered]
When im using parameters in tests, the @Epic annotation is not visible in the Behaviors tab in the allure report. 
![image](https://github.com/allure-framework/allure2/assets/107676190/7a5a35f2-20c6-43b3-82db-ced861f816f7)

The test with parameters also have a @Epic, @Feature and @Story annotation. But the report only shows the name and parameter
 @Test
    @Epic("xxx")
    @Feature("YYY")
    @Story("ZZZ'" )
    @Parameters({
            "1",
            "2"
    })

**Comment by baev** [comment]
@MM-KR, please, specify the framework and Allure integration you're using.

**Reply by MM-KR** [reply]
I'm still a beginner and trying to find my way, so I hope I understand and answer the question correctly. When I look in the POM.xml I see JUnit4 and Selenium, among others. The tests are run in Jenkins.

**Reply by baev** [reply]
For JUnit 4, all the annotations should be supported just fine. Could you please provide some additional details? The best would be to have a Github repo with a sample project reproducing the issue. Alternatively, you can provide a sample test by following the steps:

1. Remove the test body from the method
2. Remove all the other test & utility methods from the class
3. Remove all unused imports
4. Rename the class & test method to something generic. 
5. Remove all the extra annotations and metadata from the test method
6. Make sure you can run the test just fine, and the problem is still reproduced 
7. Reply with the test class content (including imports)
8. Make sure you use the GitHub code block in your reply https://docs.github.com/en/get-started/writing-on-github/working-with-advanced-formatting/creating-and-highlighting-code-blocks

**Comment by MM-KR** [comment]
```Java
package xx.xxxx.protocols.Urine.Micro;

import io.qameta.allure.Epic;
import io.qameta.allure.Feature;
import io.qameta.allure.Story;
import junitparams.JUnitParamsRunner;
import junitparams.Parameters;
import xx.xxxxx.pages.elements.Rubriek;
import xx.xxxxx.protocols.Urine.UrineBase;
import org.junit.Test;
import org.junit.runner.RunWith;

import static com.codeborne.selenide.Condition.text;
import static com.codeborne.selenide.Condition.visible;
import static nl.palga.condition.Mandatory.mandatory;

@RunWith(JUnitParamsRunner.class)
public class EpicTest extends UrineBase {
    @Test
    @Epic("Epic")
    @Feature("Feature")
    @Story("Story" )
    @Parameters({
            "weinig (< 5-10 cellen)",
            "veel (> 5-10 cellen)"
    })

    public void epicTestTest(String answersAantalAtypischeUrotheelcellen){
        //Select Micro Tab
        protocolPage.selectNavigationTab("Micro");

        //Setup Micro Tab
        Rubriek.getByLabel("Celrijkdom").setValue("celrijk");
        Rubriek.getByLabel("Beoordeelbaarheid").setValue("goed");
        Rubriek.getByLabel("Epitheliale cellen").setValue("urotheelcellen");
        Rubriek.getByLabel("Atypische urotheelcellen").setValue("ja");

        //Check Micro Tab
        Rubriek.getByLabel("Aantal atypische urotheelcellen").shouldBe(visible, mandatory);

        //Set answer
        Rubriek.getByLabel("Aantal atypische urotheelcellen").setValue(answersAantalAtypischeUrotheelcellen);

        //Check PV
        protocolPage.getReport().shouldHave(text("Aantal atypische urotheelcellen: " + answersAantalAtypischeUrotheelcellen ));
    }
}
```

**Reply by baev** [reply, marked as answer]
Thanks for your response. The issue is that `junitparams.JUnitParamsRunner` [isn't providing the information about test method annotation for parameterised tests](https://github.com/Pragmatists/JUnitParams/blob/b8647f177a2be1f37a252ec11687b0ea606b1a70/src/main/java/junitparams/internal/ParametrizedDescription.java#L25). So Allure has no access to the method annotations and that's why you have no metadata added to your test results.

Trying to add support from our side is not worth it considering [the current state of JUnit 4](https://stackoverflow.com/a/72327058/3915166) and JUnitParams extension runner (the last release in 2017)
